plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/StaticMetastoreConfig.java

```diff
@@ -95,4 +96,17 @@ public boolean isEitherThriftOrHttpMetastore()
         boolean hasThriftMetastore = metastoreUris.stream().anyMatch(uri -> "thrift".equalsIgnoreCase(uri.getScheme()));
         return hasHttpMetastore && hasThriftMetastore;
     }
+
+    public Optional<String> getCatalogName()
+    {
+        return Optional.ofNullable(catalogName);
+    }
+
+    @Config("hive.metastore.catalog-name")
```

Comment:
I would rather argue that this setting needs to land in `io.trino.plugin.hive.metastore.thrift.ThriftMetastoreConfig`

Potential suggestion for the config name:
`hive.metastore.thrift.catalog-name`


Why?
Because HTTP Thrift (Databricks Unity) doesn't have support for multiple catalogs AFAIK and there is no reason to request for it at the moment.

What changes does this imply?
Restore `io.trino.plugin.hive.metastore.thrift.ThriftMetastoreClientFactory#create` to its original form.
In `io.trino.plugin.hive.metastore.thrift.DefaultThriftMetastoreClientFactory#create(java.net.URI, java.util.Optional<java.lang.String>)` use the catalog name retrieved from `ThriftMetastoreConfig`

In `io.trino.plugin.hive.metastore.thrift.HttpThriftMetastoreClientFactory#create` just send `Optional.empty()` for the catalog name  which should leave the existing Databricks Unity Thrift client intact.

Reply:
fine with me. sorry for the confusion here @osscm

Reply:
np @anusudarsan !

@findinpath I had change like that initially, but then based on our discussion, we thought at the static metastore config level, it make sense.

Do you think, we can keep it at the `StaticMetastoreConfig` level only, as now the config is optional and if HTTPThrift is not supporting than it should be fine.

But I am ok to change it back :)

Reply:
@osscm At the time of this writing, we have different settings for the same concept:
- `hive.metastore.thrift.client.read-timeout`
- `hive.metastore.http.client.read-timeout`

Even though code wise this may create the impression of redundancy, from a user perspective this brings more clarity in my opinion.
If the user deals with http, the user is concerned only with configuration settings which make sense for http. Same thing for thrift.
At the current time, this setting makes sense only for thrift, so that is why i'm advocating using the config setting 

`hive.metastore.thrift.catalog-name`

Reply:
sure

Reply:
updated